controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteScopeTask.java

```diff
@@ -0,0 +1,158 @@
+
+/**
+ * Request handler for executing a delete operation for a ReaderGroup.
+ */
+public class DeleteScopeTask implements ScopeTask<DeleteScopeEvent> {
+    private static final TagLogger log = new TagLogger(LoggerFactory.getLogger(DeleteScopeTask.class));
+
+    private static final int PAGE_LIMIT = 1000;
+    private final StreamMetadataTasks streamMetadataTasks;
+    private final StreamMetadataStore streamMetadataStore;
+    private final KVTableMetadataStore kvtMetadataStore;
+    private final ScheduledExecutorService executor;
+
+    public DeleteScopeTask(final StreamMetadataTasks streamMetadataTasks,
+                           final StreamMetadataStore streamMetaStore,
+                           final KVTableMetadataStore kvtMetadataStore,
+                           final ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(streamMetadataTasks);
+        Preconditions.checkNotNull(streamMetaStore);
+        Preconditions.checkNotNull(kvtMetadataStore);
+        Preconditions.checkNotNull(executor);
+        this.streamMetadataStore = streamMetaStore;
+        this.streamMetadataTasks = streamMetadataTasks;
+        this.kvtMetadataStore = kvtMetadataStore;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final DeleteScopeEvent request) {
+        String scope = request.getScope();
+        long requestId = request.getRequestId();
+        final OperationContext context = streamMetadataStore.createScopeContext(scope, requestId);
+        log.debug(requestId, "Deleting {} scope recursively", scope);
+        return streamMetadataStore.checkScopeInDeletingTable(scope, context, executor).thenCompose( exists -> {
+            if (exists) {
+                return deleteScopeContent(scope, context, requestId);
+            }
+            log.info(requestId, "Skipping processing delete scope recursive for scope {} as scope" +
+                            " does not exist in deleting table", scope);
+            return CompletableFuture.completedFuture(null);
+        });
+    }
+
+    public CompletableFuture<Void> deleteScopeContent(String scopeName, OperationContext context, long requestId) {
+        List<String> readerGroupList = new ArrayList<>();
+        Iterator<Stream> iterator = listStreams(scopeName, context).asIterator();
+
+        // Seal and delete streams and add entry to RGList
+        while (iterator.hasNext()) {
+            Stream stream = iterator.next();
+            if (stream.getStreamName().startsWith(READER_GROUP_STREAM_PREFIX)) {
+                readerGroupList.add(stream.getStreamName().substring(
+                        READER_GROUP_STREAM_PREFIX.length()));
+            }
+            log.debug("Processing seal and delete stream for Stream {}", stream);
+            Futures.getThrowingException(Futures.exceptionallyExpecting(streamMetadataTasks.sealStream(scopeName, stream.getStreamName(), requestId),
```

Comment:
As I understand you already have an enhancement request to remove this blocking API here and below. Could you please reply with this comment.

Reply:
Yes, an issue is created already for the reference. Here's the link:- https://github.com/pravega/pravega/issues/6446